### PR TITLE
Add Cognito integration to the DataProvider

### PR DESCRIPTION
**Pull request by gabrielperales**

### CODE REVIEW

File: src/providers/DataProvider.ts [hunk 1]
```diff
@@ -59,10 +60,109 @@ export class DataProvider {
...
+    return {
+      data: userAttributes,
+    };
+  }
+
   public getList = async <RecordType>(
     resource: string,
     params: GetListParams
   ): Promise<GetListResult<RecordType>> => {
+    if (resource === "users") {
```
Comment (gabrielperales) [inline src/providers/DataProvider.ts:162]: I don't like this, I'll try to find a better/nicer way to implement it.

**Comment by gabrielperales** [comment]
Hi, I'm working on this bit. For now, I just have implemented getOne and getList of "users".

I'm not sure about how to implement the other methods because Admin Queries doesn't generate Update either Delete... so to add them to this repo like a generic way we probably should provide the code people needs to add to the AdminQueries API.

File: src/providers/DataProvider.ts [hunk 2]
```diff
@@ -131,6 +231,10 @@ export class DataProvider {
     resource: string,
     params: GetOneParams
   ): Promise<GetOneResult<RecordType>> => {
+    if (resource === "users") {
```
Comment (gabrielperales) [inline src/providers/DataProvider.ts:234]: I probably should change "users" by "cognitoUsers" or something like that...

**Comment by MrHertal** [comment]
Thanks @gabrielperales !
Managing admin queries is something I had in mind for a long time, but I never found the time to do it. So I'm very glad you are working on a PR and I will do my best to help you.

File: src/providers/DataProvider.ts (hunk 2 above)
Comment (MrHertal) [reply src/providers/DataProvider.ts:234]: I agree, I think `users` is too common, `cognitoUsers` seems fine

**Comment by gabrielperales** [comment]
Great!, If you can give me some direction, I'll try to keep it clean and nice ✨

**Comment by MrHertal** [comment]
I created a new issue to discuss this: https://github.com/MrHertal/react-admin-amplify/issues/36

@gabrielperales you can comment on that issue and tell me what you think 🙂

**Pull request merged**